### ISSUE TITLE
Alternator: UpdateItemSpec got exception with error code 400

### ISSUE THREAD

**Issue by Void9711** [scylladb/scylla#8125]
*Installation details*
Scylla version (or git commit hash): [4.2.3](https://hub.docker.com/layers/scylladb/scylla/4.2.3/images/sha256-9488c01d843d8465d326102a8d1533e1e379bcbd005403f5c7366513d48d1a37?context=explore)
Cluster size: 1 (developer mode)
OS (RHEL/CentOS/Ubuntu/AWS AMI): Ubuntu 20.04

Using aws sdk java
Request object: `com.amazonaws.services.dynamodbv2.document.spec.UpdateItemSpec`

```java
private final Map<String, String> nameMap = new HashMap<>() {
    {
        put("#att", "Latest");
    }
 };

private final Map<String, Object> valueMap = new HashMap<>() {
    {
        put(":val", 1);
    }
};

UpdateItemSpec updateItemSpec = new UpdateItemSpec()
                .withPrimaryKey("Offset", 1)
                .withUpdateExpression("ADD #att :val")
                .withNameMap(nameMap)
                .withValueMap(valueMap)
                .withReturnValues("UPDATED_NEW");

UpdateItemOutcome outcome = table.updateItem(updateItemSpec);
```

```
com.amazonaws.services.dynamodbv2.model.AmazonDynamoDBException: Item has invalid format: null (Service: AmazonDynamoDBv2; Status Code: 400; Error Code: ValidationException; Request ID: null)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1712) ~[aws-java-sdk-core-1.11.546.jar:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1367) ~[aws-java-sdk-core-1.11.546.jar:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1113) ~[aws-java-sdk-core-1.11.546.jar:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:770) ~[aws-java-sdk-core-1.11.546.jar:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:744) ~[aws-java-sdk-core-1.11.546.jar:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:726) ~[aws-java-sdk-core-1.11.546.jar:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:686) ~[aws-java-sdk-core-1.11.546.jar:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:668) ~[aws-java-sdk-core-1.11.546.jar:?]
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:532) ~[aws-java-sdk-core-1.11.546.jar:?]
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:512) ~[aws-java-sdk-core-1.11.546.jar:?]
	at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.doInvoke(AmazonDynamoDBClient.java:4243) ~[aws-java-sdk-dynamodb-1.11.546.jar:?]
	at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.invoke(AmazonDynamoDBClient.java:4210) ~[aws-java-sdk-dynamodb-1.11.546.jar:?]
	at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.executeUpdateItem(AmazonDynamoDBClient.java:3916) ~[aws-java-sdk-dynamodb-1.11.546.jar:?]
	at com.amazonaws.services.dynamodbv2.AmazonDynamoDBClient.updateItem(AmazonDynamoDBClient.java:3883) ~[aws-java-sdk-dynamodb-1.11.546.jar:?]
	at com.amazonaws.services.dynamodbv2.document.internal.UpdateItemImpl.doUpdateItem(UpdateItemImpl.java:102) ~[aws-java-sdk-dynamodb-1.11.546.jar:?]
	at com.amazonaws.services.dynamodbv2.document.internal.UpdateItemImpl.updateItem(UpdateItemImpl.java:86) ~[aws-java-sdk-dynamodb-1.11.546.jar:?]
...
```
The request id was regarded as a null value.

Similar issue (maybe): #7763

**Comment by nyh** [comment]
@Void9711 thanks or the report. Indeed, #7763 has exactly the same symptoms and was never backported to 4.2 (it's in the master master, and is in 4.4rc1, but not in the 4.2 and 4.3). Can you please check if you have the same issue (is your update an "ADD" operation)? If it is, we can close this issue as a duplicate. If it's a different issue, can you please try to figure out what exactly the UpdateItem operation that fails is trying to do?
In any case, I'll backport the fix to 7763 now, so it will be in the next release on the 4.2 branch as well.

**Comment by nyh** [comment]
I missed the fact that you did include the exact query in the issue, and it was an ADD. So it's definitely a duplicate #7763. So closing this issue and will backport the fix for #7763 to the 4.2 branch (and also 4.3). 4.4rc1 and nightly builds should already have this fixed - please let me know if it doesn't fix it.

**Comment by nyh** [comment]
Backported the fix for #7763. If tests go well, it will be in the next release in the 4.2 series. And as I said above, if you want and can, you can also try the fix right now by using 4.4rc1 or a nightly build - which already have it.

**Comment by Void9711** [comment]
Thanks @nyh ! Using 4.4.rc1 instead works. Actually it's a duplicate issue.
